Update client version, update tests for new retriever format and update auth flow

ssh-esh opened the pull request:
**Issues**
Closes #79 
Closes #81 

**Description**
Note: Apologies to the reviewers, in hindsight this should have been probably two separate PRs. If it becomes difficult to review then I will split it and close this PR. I will try to split the concerns by commits for easy review 

- **First commit:** update the poetry lock file(auto-generataed) and client version as part of #79 
- **Second commit:** updates auth flow and enforces usage of utility functions for clients and read env variables as part of #81 
- **Third commit:** updates query body assertion for new retriever format as part of #79 
- Rest of the commits are about keeping the linting tests happy 

The lock file changes seem quite extensive but I think it is due to poetry being upgraded from 1.84 -> 2.1.4. The old lock file is incompatible with the new client it seems so i had to do a `poetry lock` command to generate this lock file 🤔

ezimuel commented on libs/elasticsearch/tests/integration_tests/_async/test_vectorstores.py:
```diff
@@ -594,37 +594,50 @@ def assert_query(
-                    "bool": {
-                        "filter": [],
-                        "must": [{"match": {"text": {"query": "foo"}}}],
+                "retriever": {
```
Yes, I think we need to adjust the dependencies as follows:
`langchain-elastic < 0.4.0` -> `elasticsearch-py <8.16.0`
`langchain-elastic >= 0.4.0` -> `elasticsearch-py >=8.16.0`

@miguelgrinberg I think we should remove the `poetry.lock` file from the repository (adding it to `.gitignore`).

This means we need to ship another version for `langchain-elastic` 0.3.3 with the new poetry dependency:
`elasticsearch-py <8.16.0`

ssh-esh commented:
monas

> @ssh-esh The lockfile does not need to be in the repo, I agree. But removing the lockfile does not address the problem. Correct me if I'm wrong, but if you say `^8.15.1` in pyproject.toml, that means that you are asking for the latest 8.15.x release. You can change this to `^8.15` and then you would be asking for the latest 8.x.x release, which is better, but still leaves the 9.x.x versions out. My suggestion to use `>=` and `<` for bounds checking seems to match our needs better.

Yes absolutely the lock file was a separate problem and will not address the main issue here 

Based on yours and @ezimuel comments here is my understanding of next steps: 
- I make a separate PR from this and we release `langchain-elastic 0.3.3` with `elasticsearch-py <8.16.0 `(or just pin it maybe to `8.15.1`?) so that anyone  who wants to use the old flow can have the client that send the requests without retriever 
- We follow up with this PR and release `langchain-elastic 0.4.0` with  `elasticsearch-py ">=8.16 <9"` for users to use the new retriever format and remove the dead RRF code 
- We remove the `poetry lock file` in this PR 

Please let me know if I have missed anything and my understanding is clear 👍

ssh-esh commented:
Summary of changes since the last comment made by @miguelgrinberg : 
- remove of the lock file
- add lock file to git ignore 
- update elasticsearch client version range 
- consolidate dependancies from lock file into the the pyproject.toml. We are pinning versions instead of downloading latest .git as this would require us to update python dependancies and other dependancies. v1 langchain was released a few days ago
  - langchain version pin to "^0.3.10" 
  - langchain-community pin to "^0.3.10"
  - langchain-text-splitters pin to "^0.3.0"
  - Remove langchain-core from sections like [tool.poetry.group.test.dependencies] as this is already defined in the main [tool.poetry.dependencies] so it will be installed already
 - Remove a now redundant check for the lock file in the workflow inside file _lint.yml

ssh-esh commented:
> Summary of changes since the last comment made by @miguelgrinberg :
> 
> * remove of the lock file
> * add lock file to git ignore
> * update elasticsearch client version range
> * consolidate dependancies from lock file into the the pyproject.toml. We are pinning versions instead of downloading latest .git as this would require us to update python dependancies and other dependancies. v1 langchain was released a few days ago
>   
>   * langchain version pin to "^0.3.10"
>   * langchain-community pin to "^0.3.10"
>   * langchain-text-splitters pin to "^0.3.0"
>   * Remove langchain-core from sections like [tool.poetry.group.test.dependencies] as this is already defined in the main [tool.poetry.dependencies] so it will be installed already
> * Remove a now redundant check for the lock file in the workflow inside file _lint.yml

Adding more flexibility for the versions here instead of carats, i am using >= on the same version numbers and allow poetry to resolve the highest possible versions. It will not install v1 langchain as that requires a minimum python version of 3.10,right now we support 3.9. It should be enough for a 0.4.0 release and then we will support v1 langchain when we release v1 langchain-elastic 👍

ssh-esh commented:
After some feedback the last few commits will 
- update langchain version range to langchain = ">=0.3.10,<1.0.0"
-  remove langchain_community and langchain_text_splitters from the test dependancies. community can be substituted in our usage by langchain_core and text splitters is automatically installed with 'langchain'
- Use langchain_core for FakeMEssagesListChatModel in tests
- add DistanceStrategy import to __init__
- update spynhix docs in ElasticSearchStore to not use langchain_community
